src/Components/TopContent.tsx

```diff
@@ -1,117 +1,174 @@
-import React from 'react'
-import { Menubar } from 'primereact/menubar';
-import { MenuItem } from 'primereact/menuitem';
-import { useNavigate } from 'react-router-dom';
-import { Divider } from 'primereact/divider';
-import { InputText } from 'primereact/inputtext';
-import { Button } from 'primereact/button';
-import { Card } from 'primereact/card';
-export default function TopContent() {
-
-    const navigate = useNavigate()
-
-    const items: MenuItem[] = [
+import React, { useState } from "react";
+import { Menubar } from "primereact/menubar";
+import { MenuItem } from "primereact/menuitem";
+import { useNavigate } from "react-router-dom";
+import { ConfirmDialog, confirmDialog } from "primereact/confirmdialog";
+import { Divider } from "primereact/divider";
+import { InputText } from "primereact/inputtext";
+import { Button } from "primereact/button";
+import { Card } from "primereact/card";
+import { ListBox, ListBoxChangeEvent } from "primereact/listbox";
+
+interface Libelle {
+  name: string;
+  code: string;
+}
+interface Product {
```

Comment:
we can make new folder /src/models/model_name (Product here) to more organize your code.